upload-api/buckets/delegations-store.js

```diff
@@ -69,7 +69,7 @@ export const useDelegationsStore = (s3client, bucketName) => {
         Bucket: bucketName,
         Key: createDelegationsBucketKey(cid),
       })
-      const s3Object = await s3client.send(getObjectCmd)
+      const s3Object = await pRetry(() => s3client.send(getObjectCmd))
```

Comment:
I assume it is possible to have 404? If so, we should have something more in the lines of

```js
    const s3Object = await pRetry(async () => {
    let res
      try {
        res = await this.client.send(cmd)
      } catch (cause) {
        if (cause?.$metadata?.httpStatusCode === 404) {
          return undefined
        }
        throw new Error(`Failed to check bucket ${this.name}`)
      }
      return res
    }, {
      retries: 3
    })
```